cylc/flow/taskdef.py

```diff
@@ -88,9 +82,24 @@ def add_dependency(self, dependency, sequence):
                 this dependency applies.
 
         """
-        if sequence not in self.dependencies:
-            self.dependencies[sequence] = []
-        self.dependencies[sequence].append(dependency)
+        try:
+            self.dependencies[sequence].append(dependency)
+        except KeyError:
+            self.dependencies[sequence] = [dependency]
+
+    def add_xtrig_label(self, xtrig_label, sequence):
+        """Add an xtrigger to a named sequence.
+
+        Args:
+            xtrig_label: The xtrigger label to add.
+            sequence (cylc.cycling.SequenceBase): The sequence for which this
+                xtrigger applies.
+
+        """
+        try:
+            self.xtrig_labels[sequence].append(xtrig_label)
+        except KeyError:
+            self.xtrig_labels[sequence] = [xtrig_label]
```

Comment:
Ditto the above for replacing try/except with `setdefault`

Reply:
:+1: